felix/dataplane/linux/vxlan_mgr_test.go

```diff
@@ -30,18 +30,34 @@ import (
 )
 
 type mockVXLANDataplane struct {
-	links []netlink.Link
+	links     []netlink.Link
+	ipVersion uint8
 }
 
 func (m *mockVXLANDataplane) LinkByName(name string) (netlink.Link, error) {
-	return &netlink.Vxlan{
+	link := &netlink.Vxlan{
 		LinkAttrs: netlink.LinkAttrs{
 			Name: "vxlan",
 		},
 		VxlanId:      1,
 		Port:         20,
 		VtepDevIndex: 2,
-		SrcAddr:      ip.FromString("172.0.0.2").AsNetIP()}, nil
+		SrcAddr:      ip.FromString("172.0.0.2").AsNetIP(),
+	}
+
+	if m.ipVersion == 6 {
+		link = &netlink.Vxlan{
+			LinkAttrs: netlink.LinkAttrs{
+				Name: "vxlan",
```

Comment:
question: do we need this to be different for v6 and v4? (not sure at all, just wondering, feel free to ignore if this doesn't make sense heh)

Reply:
I need to update this to match the previous changes for changing the device ID to `vxlan-v6.calico`. It works for the tests (since they don't even have the `.calico` portion we use in reality) but I should make it match a little bit better.